sql/parser/builtins.go

```diff
@@ -399,15 +401,46 @@ var builtins = map[string][]builtin{
 		builtin{
 			types:      typeList{},
 			returnType: DummyDate,
+			fn: func(e EvalContext, args DTuple) (Datum, error) {
+				return DDate{Time: e.StmtTimestamp.Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"statement_timestamp": {nowImpl},
+	"current_timestamp":   {nowImpl},
+	"now":                 {nowImpl},
+
+	"clock_timestamp": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyTimestamp,
 			impure:     true,
 			fn: func(_ EvalContext, args DTuple) (Datum, error) {
-				return DDate{Time: time.Now().Truncate(24 * time.Hour)}, nil
+				return DTimestamp{Time: time.Now()}, nil
 			},
 		},
 	},
 
-	"current_timestamp": {nowImpl},
-	"now":               {nowImpl},
+	"transaction_timestamp": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyTimestamp,
+			fn: func(e EvalContext, args DTuple) (Datum, error) {
+				return DTimestamp{Time: e.TxnTimestamp}, nil
+			},
+		},
+	},
+
+	"transaction_date": {
```

Comment:
Does any other database have a `transaction_date()` function? I'm having a hard time seeing a use for it and it's always expressible as `cast(transaction_timestamp() as date)`.